tools/src/launcher/Main.hs

```diff
@@ -586,8 +589,13 @@ writeWindowsUpdaterRunner :: FilePath -> M ()
 writeWindowsUpdaterRunner runnerPath = liftIO $ do
     exePath <- getExecutablePath
     launcherArgs <- getArgs
+#ifdef mingw32_HOST_OS
+    selfPid <- Process.getCurrentProcessId
+#else
+    let selfPid = 0 -- This will never be run on non-Windows
+#endif
     writeFile (toString runnerPath) $ unlines
-        [ "TaskKill /IM cardano-launcher.exe /F"
+        [ "TaskKill /PID "<>show selfPid<>" /F"
```

Comment:
Not a big fan of this to be honest, and the pragma on top the file... And, what the comment truly says is that, if someone ever needs the windows update runner, it means they've ran the launcher on windows first, and thus, that those lines have been ran on windows? Is that correct? 

I mean, this code still runs on linux and darwin machines, right? It's just that we make no use of the "update runner"...

Reply:
@KtorZ, the `runnerPath = udWindowsPath ud :: Maybe FilePath` governs whether `writeWindowsUpdaterRunner` will be run, and is naturally `Nothing` on Linux.  We can fix the type defaulting, of course, if that's critical.

Reply:
To clarify, in other circumstances I'd not resort to `-Wno-type-default`, but I needed a sure-fire way to get past the thing quick -- and I don't have ready access to a Windows machine..

..and this is a release blocker..

Reply:
Yeah, got you (hence why I haven't blocked anything on this PR). I was just wondering. One way to avoid the warning would be to go for a string right away `show <$>  Process.getCurrentProcessId`, but that's really minor.